babel/src/test/java/org/apache/calcite/test/BabelParserTest.java

```diff
@@ -974,6 +974,185 @@ private void checkParseInfixCast(String sqlType) {
+        + "return current_date + 1";
+    final String expected = "CREATE FUNCTION `FOO` () "
+        + "RETURNS INTEGER "
+        + "LANGUAGE SQL "
+        + "COLLATION INVOKER INLINE TYPE 1 "
+        + "RETURN (CURRENT_DATE + 1)";
+    sql(sql).ok(expected);
+  }
+
+  @Test public void testCreateFunctionSqlFormOneParameter() {
+    final String sql =
+        "create function foo(a Integer) "
+            + "returns Integer "
+            + "language sql "
+            + "collation invoker inline type 1 "
+            + "return current_date + 1";
+    final String expected = "CREATE FUNCTION `FOO` (`A` INTEGER) "
+        + "RETURNS INTEGER "
+        + "LANGUAGE SQL "
+        + "COLLATION INVOKER INLINE TYPE 1 "
+        + "RETURN (CURRENT_DATE + 1)";
+    sql(sql).ok(expected);
+  }
+
+  @Test public void testCreateFunctionSqlFormNotIncludeLanguageSQLFails() {
+    final String sql =
+        "create function foo() "
+            + "returns Integer "
+            + "^collation^ invoker inline type 1 "
+            + "return current_date + 1";
+    final String expected = "(?s).*Encountered \"collation\" at .*";
+    sql(sql).fails(expected);
+  }
+
+  @Test public void testCreateFunctionSqlFormMoreThanOneParameter() {
+    final String sql =
+        "create function foo(a Integer, b varchar(30) ) "
+            + "returns Integer "
+            + "language sql "
+            + "collation invoker inline type 1 "
+            + "return current_date + 1";
+    final String expected = "CREATE FUNCTION `FOO` (`A` INTEGER, `B` VARCHAR(30)) "
+        + "RETURNS INTEGER "
+        + "LANGUAGE SQL "
+        + "COLLATION INVOKER INLINE TYPE 1 "
+        + "RETURN (CURRENT_DATE + 1)";
+    sql(sql).ok(expected);
+  }
+
+  @Test public void testCreateFunctionSqlFormNotDeterministic() {
+    final String sql =
+        "create function add1(a Integer) "
+            + "returns Integer "
+            + "language sql "
+            + "not deterministic "
+            + "collation invoker inline type 1 "
+            + "return a + 1";
+    final String expected = "CREATE FUNCTION `ADD1` (`A` INTEGER) "
+        + "RETURNS INTEGER "
+        + "LANGUAGE SQL "
+        + "NOT DETERMINISTIC "
+        + "COLLATION INVOKER INLINE TYPE 1 "
+        + "RETURN (`A` + 1)";
+    sql(sql).ok(expected);
+  }
+
+  @Test public void testCreateFunctionSqlFormDeterministic() {
+    final String sql =
+        "create function add1(a Integer) "
+            + "returns Integer "
+            + "language sql "
+            + "deterministic "
+            + "collation invoker inline type 1 "
+            + "return a + 1";
+    final String expected = "CREATE FUNCTION `ADD1` (`A` INTEGER) "
+        + "RETURNS INTEGER "
+        + "LANGUAGE SQL "
+        + "DETERMINISTIC "
+        + "COLLATION INVOKER INLINE TYPE 1 "
+        + "RETURN (`A` + 1)";
+    sql(sql).ok(expected);
+  }
+
+  @Test public void testCreateFunctionSqlFormCalledOnNUllInput() {
```

Comment:
nit: capitalization in this method and the one below, "NU" in Null

Reply:
Fixes made in https://github.com/googleinterns/calcite/pull/79/commits/973cb3f514eabcc9dcb2e51bebf64c3d23d9e7c0 . Thank you for pointing it out.